framework/includes/option-types/popup/static/js/popup.js

```diff
@@ -58,12 +58,33 @@
 			},
 			'open': function () {
 				$this.trigger('fw:option-type:popup:open');
+
+				if (data['custom-events']['open']) {
+					fwEvents.trigger(data['custom-events']['open'], {
```

Comment:
To prevent conflicts, maybe force a prefix for event name, let's say

```php
fwEvents.trigger('fw:option:popup:custom-event:'+ data['custom-events']['open']);
```

_maybe you know a better prefix name_

?